app/controllers/application_controller.rb

```diff
@@ -2,4 +2,17 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
```

Comment:
It would be better to add if session[:user_id] to the end of this to avoid running the query when we know the session[:user_id] is nil.